### ISSUE TITLE
Retroachievements fail to trigger on autosave load

### ISSUE THREAD

**Issue by brunocar** [spruceUI/spruceOS#44]
After doing more testing i've figured out that retroachievements have a problem across the board, that gets worse when playing more CPU intensive games, where using the auto resume feature will cause them to fail to connect to retroachievements, requiring you exit back out and try again.

basically the connection to RAC has 2 stages, a login (which on success shows a notification on the bottom left that you are logged into your account successfully) and game approval (where the API checks the hash reported by RA and sends back a confirmation, denoted by a notification on the top left with the name of the game, an icon and the number of achievements).

2.2.2 adds a tweak to the auto resume which checks whether the wifi is connected before starting the game, after testing, this is objectively an improvement but it leaves room for improvement, see, checking that the wifi is connected guarantees that the login notif will pop up correctly, but if your internet isnt fast or stable enough (and the a30 is quite fickle about it), it'll start the game before the RAC API answers back, failing to trigger the second notification.

in short, if possible, it'd be good to also add a check for the full OK from the API.

that aside i still havent been able to track why retroarch just decides to crash if you open the achievements menu before the icon data is fully cached, this is not retroarch's normal behavior for this.

**Comment by Sundownersport** [comment]
After much discussion with the team this is our suggestion.

do not use the quick save functionality for these systems if you want RAC to function 100% of the time.

the current method of waiting to connect is as good as it’s gonna get in the foreseeable future.
